### PR TITLE
hand written IstioEndpoint deepcopy

### LINKED ISSUE #52446: Move off `copystructure`
We are using hand written DeepCopy methods in most places, but use copystructure for IstioEndpoints. This _probably_ has a performance impact (untested!), so we should consider moving off.

```
pkg/test/framework/components/echo/config.go
23:     "github.com/mitchellh/copystructure"
556:    copied, err := copystructure.Copy(v)
558:            // There are 2 locations where errors are generated in copystructure.Copy:

pilot/pkg/model/service.go
35:     "github.com/mitchellh/copystructure"
1507:   copied, err := copystructure.Copy(v)
```

Test usage isn't bad, though dropping the dep entirely may make it worth it. IstioEndpoint is critical path though.

### PR DESCRIPTION
**Pull request by wulianglongrd**
fix #52446 

benchmark result:
```
goos: darwin
goarch: amd64
pkg: istio.io/istio/pilot/pkg/model
cpu: Intel(R) Core(TM) i7-9750H CPU @ 2.60GHz
                    │    old.txt     │               new.txt               │
                    │     sec/op     │    sec/op     vs base               │
EndpointDeepCopy-12   14235.5n ± 16%   271.2n ± 10%  -98.10% (p=0.002 n=6)

                    │   old.txt   │              new.txt              │
                    │    B/op     │    B/op     vs base               │
EndpointDeepCopy-12   6688.0 ± 0%   624.0 ± 0%  -90.67% (p=0.002 n=6)

                    │   old.txt    │              new.txt              │
                    │  allocs/op   │ allocs/op   vs base               │
EndpointDeepCopy-12   189.000 ± 0%   4.000 ± 0%  -97.88% (p=0.002 n=6)
```

### CODE REVIEW

**Review by howardjohn — COMMENTED** [review]
wow a lot faster than I expected! and simple. nice! 
would be good to run https://github.com/istio/istio/pull/52447 against this to verify (I can do it tomorrow if you have issues)

File: pilot/pkg/model/service.go [hunk 1]
```diff
@@ -1441,7 +1440,14 @@ func (s *Service) Equals(other *Service) bool {
 
 // DeepCopy creates a clone of IstioEndpoint.
 func (ep *IstioEndpoint) DeepCopy() *IstioEndpoint {
-	return copyInternal(ep).(*IstioEndpoint)
+	out := *ep
+	out.Labels = maps.Clone(ep.Labels)
+	out.Addresses = slices.Clone(ep.Addresses)
```
Comment (hzxuzhonghu) [inline pilot/pkg/model/service.go:1445]: seems this is not need, we donot mutate it later after created.

**Comment by zirain** [comment]
should we use `// +k8s:deepcopy-gen=true` go generate it instead of hand writting?

File: pilot/pkg/model/service.go (hunk 1 above)
Comment (wulianglongrd) [reply pilot/pkg/model/service.go:1449]: I think it might be better to keep the DeepCopy semantics here.

**Comment by wulianglongrd** [comment]
> should we use `// +k8s:deepcopy-gen=true` to generate it instead of hand writting?

I am not familiar with this usage, can it be used for any structure? In istio, it is common to write DeepCopy by hand.

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**